version calc breaks if nuget package hasn't been pushed yet

calculating version for DotnetResource.TestApplication...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x65b4eb]

goroutine 1 [running]:
github.com/miclip/dotnet-resource/out.GenerateNextVersion(0xc000086590, 0xd, 0xc0000865a0, 0x10, 0xc00009e080, 0x36, 0xc0000be180, 0x24, 0x258, 0x0, ...)
	/go/src/github.com/miclip/dotnet-resource/out/out.go:134 +0x10b
github.com/miclip/dotnet-resource/out.Execute(0xc000086590, 0xd, 0xc0000865a0, 0x10, 0xc00009e080, 0x36, 0xc0000be180, 0x24, 0x258, 0x0, ...)
	/go/src/github.com/miclip/dotnet-resource/out/out.go:65 +0x1535
main.main()
	/go/src/github.com/miclip/dotnet-resource/cmd/out/main.go:22 +0x188

After pushing manually then the versions are generated fine.

--- a/out/out.go
+++ b/out/out.go
@@ -1,16 +1,16 @@
 package out
 
 import (
-	"time"
-	"bytes"
-	"encoding/xml"
 	"bufio"
+	"bytes"
 	"context"
+	"encoding/xml"
 	"fmt"
 	"os"
 	"os/exec"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/miclip/dotnet-resource"
 	"github.com/miclip/dotnet-resource/nuget"
@@ -68,8 +68,8 @@ func Execute(request Request, sourceDir string) (Response, []byte, error) {
 			}
 		}
 		response.Metadata = append(response.Metadata, dotnetresource.MetadataPair{
-				Name:  p.PackageID,
-				Value: version,			
+			Name:  p.PackageID,
+			Value: version,
 		})
 		if strings.ToLower(request.Params.PackageType) == "library" {
 			dotnetresource.Sayf("dotnet pack for %s...\n", p.PackageID)
@@ -86,16 +86,16 @@ func Execute(request Request, sourceDir string) (Response, []byte, error) {
 			if err != nil {
 				return response, out, err
 			}
-			
+
 			nuspec := createNupsec(request, p, version)
 			enc, err := xml.Marshal(nuspec)
 			enc = []byte(xml.Header + string(enc))
-			if err!= nil {
-				dotnetresource.Fatal("error creating nuspec file",err)
+			if err != nil {
+				dotnetresource.Fatal("error creating nuspec file", err)
 			}
-			
+
 			reader := bytes.NewReader(enc)
-			client.AddFileToPackage(p.PackageID, version, p.PackageID + ".nuspec", reader)			
+			client.AddFileToPackage(p.PackageID, version, p.PackageID+".nuspec", reader)
 
 			dotnetresource.Sayf("manual pack for %s...\n", p.PackageID)
 			packOut, err := client.ManualPack(p.PackageID, version)
@@ -118,14 +118,14 @@ func Execute(request Request, sourceDir string) (Response, []byte, error) {
 
 func createNupsec(request Request, p ProjectMetaData, version string) nuget.Nuspec {
 	nugetclient := nuget.NewNugetClient(request.Source.NugetSource)
-	return nugetclient.CreateNuspec(p.PackageID,version, p.Author,p.Description,p.Owner)
+	return nugetclient.CreateNuspec(p.PackageID, version, p.Author, p.Description, p.Owner)
 }
 
 // GenerateNextVersion ....
 func GenerateNextVersion(request Request, packageName string) (string, error) {
 	nugetclient := nuget.NewNugetClient(request.Source.NugetSource)
 	pv, err := nugetclient.GetPackageVersion(context.Background(), packageName, true)
-	if err != nil {
+	if err != nil || pv == nil {
 		pv = &nuget.PackageVersion{
 			ID:      packageName,
 			Version: strings.Replace(request.Params.Version, "*", "0", -1),

